### PR TITLE
fix(lambda) File uploads

### LINKED ISSUE #4430: Lambda: isBase64encoded conversion breaks file upload
**General**
apollo-server-lambda 2.16.0
Related: https://github.com/apollographql/apollo-server/pull/4311

**Expected**
I can upload images with a multipart form data request

**Actual**
Image upload does not work anymore since this version because of the base64 decoding. I am getting

```
{
    "errorType": "Runtime.UnhandledPromiseRejection",
    "errorMessage": "TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received an instance of Object",
    "reason": [
        {
            "errorType": "TypeError",
            "errorMessage": "The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received an instance of Object",
            "message": "The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received an instance of Object",
            "extensions": {
                "code": "INTERNAL_SERVER_ERROR"
            }
        }
    ],
    "promise": {},
    "stack": [
        "Runtime.UnhandledPromiseRejection: TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received an instance of Object",
        "    at process.<anonymous> (/var/runtime/index.js:35:15)",
        "    at process.emit (events.js:310:20)",
        "    at process.EventEmitter.emit (domain.js:482:12)",
        "    at processPromiseRejections (internal/process/promises.js:209:33)",
        "    at processTicksAndRejections (internal/process/task_queues.js:98:32)"
    ]
}

```

**Code**
```
export default async function defaultHandler(
  obj: undefined,
  args: Record<string, any>,
  context: ApolloContext,
  info: GraphQLResolveInfo,
): Promise<any> {
  let { body } = context;
  let { headers } = context;
  if (context.headers['content-type'].startsWith('multipart/form-data')) {
    body = new FormData();

    //  Await the file, we always use content as variable
    const file: Upload = ...;

    const operations = {
      ...context.body,
      variables: { ...context.body.variables, content: null },
    };

    body.append('operations', JSON.stringify(operations));
    body.append('map', JSON.stringify({ '0': ['variables.content'] }));
    body.append('0', file.createReadStream(), file);

  }
```

Had to switch back to v2.15.1 for now

### PR DESCRIPTION
**Pull request by mathix420**
Update `graphqlHandler.ts` to ignore base64 decoding for Multipart queries

Fix #4430, caused by #4311.

### CODE REVIEW

**Comment by mathix420** [comment]
Been using this fix in production since then no more issues.

**Comment by mathix420** [comment]
@abernix @glasser tell me if anything is missing for this PR, I have some time.

File: packages/apollo-server-lambda/src/lambdaApollo.ts [hunk 1]
```diff
@@ -32,9 +32,12 @@ export function graphqlLambda(
     callback,
   ): void => {
     context.callbackWaitsForEmptyEventLoop = false;
-    let { body, isBase64Encoded } = event;
+    let { body, headers, isBase64Encoded } = event;
+    let query: Record<string, any> | Record<string, any>[];
+    const contentType = headers["content-type"] || headers["Content-Type"];
+    const isMultipart = (contentType || "").startsWith("multipart/form-data");
```
Comment (ivanmartos) [inline packages/apollo-server-lambda/src/lambdaApollo.ts:38]: I would suggest to lowercase `contentType` just to be sure

**Comment by glasser** [comment]
Hi, I'm not an active maintainer of this project. Sorry!

File: packages/apollo-server-lambda/src/lambdaApollo.ts (hunk 1 above)
Comment (mathix420) [reply packages/apollo-server-lambda/src/lambdaApollo.ts]: Done, thanks for the review.
Do you know how to help this PR to be merged faster?

Comment (ivanmartos) [reply packages/apollo-server-lambda/src/lambdaApollo.ts]: sorry :(

**Review by abernix — APPROVED** [review]

**Pull request merged**